packages/snaps-execution-environments/src/offscreen/OffscreenSnapExecutor.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  BasePostMessageStream,
+  WindowPostMessageStream,
+} from '@metamask/post-message-stream';
+import { createWindow } from '@metamask/snaps-utils';
+import { JsonRpcParams, JsonRpcRequest, assert } from '@metamask/utils';
+
+type ExecutorJob = {
+  id: string;
+  window: Window;
+  stream: WindowPostMessageStream;
+};
+
+export class OffscreenSnapExecutor {
+  readonly #stream: BasePostMessageStream;
+
+  readonly jobs: Record<string, ExecutorJob> = {};
+
+  /**
+   * Initialize the executor with the given stream. This is a wrapper around the
```

Comment:
Perhaps explain that this isn't a traditional executor? Not sure if we should call it something else actually?

It is more an execution service than an executor 😅

Reply:
Perhaps `OffscreenSnapProxyExecutor`, to make it clear that it doesn't execute things itself?

Reply:
Potentially yeah. I would leave an extensive comment/doc string about it as well